models/addresses_summary/addresses_summary_op_mainnet.sql

```diff
@@ -0,0 +1,216 @@
+           WHEN date_diff('day', max(ot.block_time), CAST(NOW() as timestamp)) > 365 THEN 'Last active more than 1 Year Ago'
+           WHEN date_diff('day', max(ot.block_time), CAST(NOW() as timestamp)) > 91 THEN 'Last active more than 3 months Ago'
+           WHEN date_diff('day', max(ot.block_time), CAST(NOW() as timestamp)) > 30 THEN 'Last active more than 1 month Ago'
+           WHEN date_diff('day', max(ot.block_time), CAST(NOW() as timestamp)) > 7 THEN 'Last active more than 1 week Ago'
+           ELSE 'Active within the past week'
+    END as recency_age, 
+    date_diff('day', min(ot.block_time), max(ot.block_time)) as address_age_in_days,
+    date_diff('day', max(ot.block_time), CAST(NOW() as timestamp)) as recency_in_days, 
+    CASE
+            WHEN (COUNT(ot.hash)/(date_diff('day', min(ot.block_time), max(ot.block_time)) + 1)) >= 1 THEN 'Daily User'
+            WHEN (COUNT(ot.hash)/(date_diff('day', min(ot.block_time), max(ot.block_time)) + 1)) >= 0.142857142857 THEN 'Weekly User'
+            WHEN (COUNT(ot.hash)/(date_diff('day', min(ot.block_time), max(ot.block_time)) + 1)) >= 0.0333333333333 THEN 'Monthly User'
+            WHEN (COUNT(ot.hash)/(date_diff('day', min(ot.block_time), max(ot.block_time)) + 1)) >= 0.0027397260274 THEN 'Yearly User'
+            ELSE 'Sparse User'
+    END as usage_frequency,
+    COUNT(ot.hash) as number_of_transactions,
+    COUNT(DISTINCT(cm.contract_project)) as unique_dapps,
+    MIN_BY(cm.contract_project, ot.block_number) as first_to_project, 
+    MIN_BY(ot.to, ot.block_number) as first_to_address
+FROM 
+weekly_active_addresses wd 
+INNER JOIN (
+        SELECT 
+            "from",
+            to, 
+            hash,
+            block_time, 
+            (l1_fee + (gas_used * gas_price))/1e18 as gas_spent,
+            data, 
+            block_number
+        FROM 
+        {{ source('optimism', 'transactions') }}
+
+        UNION ALL 
+
+        SELECT 
+            "from",
+            to,
+            hash,
+            block_time,
+            case when gas_price = 0 then 0
+            else cast(gas_limit as double) * cast(gas_price as double)/1e18
+            end as gas_spent,
+            data,
+            block_number
+        FROM 
+        {{ source('optimism_legacy_ovm1', 'transactions') }}
+) ot 
+    ON wd.address = ot."from"
+INNER JOIN 
+{{ ref('addresses_events_optimism_first_activity') }} fa 
+    ON wd.address = fa.address
+LEFT JOIN 
+{{ ref('contracts_optimism_contract_mapping') }} cm 
+    ON ot."to" = cm.contract_address 
+LEFT JOIN (
+    SELECT 
+        DISTINCT id, 
+        split_part(signature,'(',1) as function 
+    FROM 
+    {{ ref('signatures') }} 
+    where type = 'function_call'
+    AND id NOT IN (0x09779838, 0x00000000) -- for some weird reason these have duplicates functions
+) sig 
+    ON sig.id = fa.first_function
```

Comment:
maybe this part can be done in `addresses_events_optimism_first_activity` instead of here?